bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs

```diff
@@ -417,20 +419,24 @@ pub mod pallet {
 		}
 
 		/// Process a delivery receipt from a relayer, to allocate the relayer reward.
-		pub fn process_delivery_receipt(
-			relayer: <T as frame_system::Config>::AccountId,
-			receipt: DeliveryReceipt,
-		) -> DispatchResult {
+		pub fn process_delivery_receipt(receipt: DeliveryReceipt) -> DispatchResult {
 			// Verify that the message was submitted from the known Gateway contract
 			ensure!(T::GatewayAddress::get() == receipt.gateway, Error::<T>::InvalidGateway);
 
+			let reward_account = T::AccountId::decode(&mut receipt.reward_address.as_slice())
```

Comment:
I'm sure `T::AccountId` supports `From<[u8; 32]>`. That's more semantically correct than using `T::AccountId::decode` which only works in this case by incidence.

Reply:
Also converting `[u8; 32]` to AccountId is _infallible_. it should not fail, so we will never ever raise `InvalidRewardAccount` in practise.

Reply:
> I'm sure T::AccountId supports From<[u8; 32]>. That's more semantically correct than using T::AccountId::decode which only works in this case by incidence.

From<[u8; 32]> for T::AccountId does not work.

Reply:
Yeah, `AccountId` is generic and it's better not to discard its generic nature by explicitly declaring `From<[u8; 32]>` in the pallet definition.

https://substrate.stackexchange.com/a/1318/523

Reply:
> From<[u8; 32]> for T::AccountId does not work.

https://github.com/yrong/polkadot-sdk/pull/18

> Yeah, AccountId is generic and it's better not to discard its generic nature by explicitly declaring From<[u8; 32]> in the pallet definition.

This does not make sense. This pallet explicitly requires that a 32-byte sequence be directly converted to a `T::AccountId`, and so `From<[u8; 32]>` is the right constraint.